modules/Microsoft.HealthDataServices/fhirservices/deploy.bicep

```
@@ -0,0 +1,298 @@
+  properties: {
+    accessPolicies: accessPolicies
+    authenticationConfiguration: {
+      authority: authenticationAuthority
+      audience: authenticationAudience
+      smartProxyEnabled: smartProxyEnabled
+    }
+    corsConfiguration: {
+      allowCredentials: corsAllowCredentials
+      headers: corsHeaders
+      maxAge: corsMaxAge == -1 ? null : corsMaxAge
+      methods: corsMethods
+      origins: corsOrigins
+    }
+    publicNetworkAccess: publicNetworkAccess
+    exportConfiguration: exportStorageAccountName == '' ? {} : exportConfiguration
+    importConfiguration: {
+      enabled: importEnabled
+      initialImportMode: initialImportMode
+      integrationDataStore: importStorageAccountName == '' ? null : importStorageAccountName
+    }
+    resourceVersionPolicyConfiguration: {
+      default: resourceVersionPolicy
+      resourceTypeOverrides: empty(resourceVersionOverrides) ? null : resourceVersionOverrides
+    }
+    acrConfiguration: {
+      loginServers: acrLoginServers
+      ociArtifacts: empty(acrOciArtifacts) ? null : acrOciArtifacts
+    }
+  }
+}
+
+resource fhir_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock)) {
+  name: '${fhir.name}-${lock}-lock'
+  properties: {
+    level: any(lock)
+    notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: fhir
+}
+
+resource fhir_diagnosticSettings 'Microsoft.Insights/diagnosticsettings@2021-05-01-preview' = if ((!empty(diagnosticStorageAccountId)) || (!empty(diagnosticWorkspaceId)) || (!empty(diagnosticEventHubAuthorizationRuleId)) || (!empty(diagnosticEventHubName))) {
+  name: diagnosticSettingsName
+  properties: {
+    storageAccountId: !empty(diagnosticStorageAccountId) ? diagnosticStorageAccountId : null
+    workspaceId: !empty(diagnosticWorkspaceId) ? diagnosticWorkspaceId : null
+    eventHubAuthorizationRuleId: !empty(diagnosticEventHubAuthorizationRuleId) ? diagnosticEventHubAuthorizationRuleId : null
+    eventHubName: !empty(diagnosticEventHubName) ? diagnosticEventHubName : null
+    metrics: diagnosticsMetrics
+    logs: diagnosticsLogs
+  }
+  scope: fhir
+}
+
+
+module fhir_roleAssignments '.bicep/nested_roleAssignments.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${deployment().name}-Rbac-${index}'
+  params: {
+    description: contains(roleAssignment, 'description') ? roleAssignment.description : ''
+    principalIds: roleAssignment.principalIds
+    principalType: contains(roleAssignment, 'principalType') ? roleAssignment.principalType : ''
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    condition: contains(roleAssignment, 'condition') ? roleAssignment.condition : ''
+    delegatedManagedIdentityResourceId: contains(roleAssignment, 'delegatedManagedIdentityResourceId') ? roleAssignment.delegatedManagedIdentityResourceId : ''
+    resourceName: '${workspaceName}/${fhir.name}'
```

```suggestion
    resourceId: fhir.id
```

Allows us to be more dynamic with less resource-specific code.